src/react.mjs

```diff
@@ -0,0 +1,60 @@
+import globals from 'globals';
+import testingLibrary from 'eslint-plugin-testing-library';
+import hooks from 'eslint-plugin-react-hooks';
+import react from 'eslint-plugin-react';
+import cypress from 'eslint-plugin-cypress';
+
+import baseConfig from './index.mjs'; // Import the base config
+
+export default [
+  ...baseConfig,
+  {
+    languageOptions: {
+      globals: {
+        ...globals.node,
+        ...globals.browser,
+      },
+    },
+    plugins: {
+      react,
+      'react-hooks': hooks,
+      'testing-library': testingLibrary,
+      cypress,
+    },
+    settings: {
+      react: {
+        version: 'detect',
+      },
+    },
+    rules: {
+      'react-hooks/exhaustive-deps': 'error',
```

Comment:
I recall that sonar included lint rules for hooks. Can we validate that we don't duplicate the rules?

Reply:
@braopa-smg

Reply:
Not a duplicate here - validated by adding a bad NoExhaustiveDeps test (and kept it) - removing the rule does not catch it ( `'react-hooks/exhaustive-deps'` ) and I don't see a specific sonar rule that would replace it